internal/server/handlers.go

```diff
@@ -27,7 +27,12 @@ func (s *Server) Subscribe(c *gin.Context) {
 	subscription := models.Email{Email: subscriptionData.Email, Status: models.Subscribed}
 	result := database.DB.FirstOrCreate(&subscription, models.Email{Email: subscriptionData.Email})
 
-	if result.Error != nil || (result.RowsAffected == 0 && subscription.Status == models.Subscribed) {
+	if result.Error != nil {
+		c.Status(http.StatusInternalServerError)
```

Comment:
I think it's worth adding error logging, because otherwise we won't know what went wrong

Reply:
Got it, it is without any content due to the contracts we had for the test case, will update it, thanks!